### DISCUSSION TITLE
Get named Instances

### DISCUSSION THREAD

**Discussion by julianhahn** [pmndrs/zustand#1927] [Q&A]
Hello everyone, I scrolled through the docs and hopefully, I did not just overread it, but is there a way to set the name of a store for the redux devtools? We decided to have multiple stores for components on different hierarchy levels.
An App Store, a Mode Store and a Flow Store for example.

But currently, just one store gets a name assigned, for which it is not clear where it came from and the stores after that get the name "instance" + counter.

Since the first one got a name that is in relation to the project, there is a functionality to set the name. 
Hope you could help me out here.

(And I'm not sure why it shows up multiple times)

<img width="720" alt="Screenshot 2023-07-10 at 1 57 42 PM" src="https://github.com/pmndrs/zustand/assets/40185075/e7d5650a-f6cc-4ff5-8d73-210bea14c98a">

**Comment by dbritto-dev** [comment]
@julianhahn would you mind sharing us how you are setting up your stores for devtools?

**Reply by julianhahn** [reply]
sorry for the delay and thanks for the fast answer!
Had to pull out the code, as I cannot share the whole project.

Again, the idea is, that we have modes and depending on the modes we switch the components in app. Each Mode in itself has different component and a bit completed state, so each mode should have it's own store. 
If you open the redux devtools on the following link, you'll see some weird behaviour.
any tips and explanations are very welcome!

https://975zlh-3000.csb.app/

https://codesandbox.io/p/github/julianhahn/zustand_named_devtools/main?workspaceId=43c04159-6486-4858-bb77-1bd4b68b823f

**Reply by julianhahn** [reply]
@dbritto-dev  could the example clarify my problem?

**Reply by dbritto-dev** [reply]
I'll take a look of that tomorrow, sorry for the late response

**Reply by dbritto-dev** [reply]
@julianhahn your codesandbox demo just fail

![image](https://github.com/pmndrs/zustand/assets/1697714/be88e8b3-533e-4a05-94e1-b853ca5ebef4)

**Reply by julianhahn** [reply]
sorry, that's weird.
Just created a shareable link to the sandbox instance itself. Tested it out in incognito mode in chrome.
[this](https://codesandbox.io/p/github/julianhahn/zustand_named_devtools/main?layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clk5bdx87000g3b6qth228mva%2522%252C%2522sizes%2522%253A%255B70%252C30%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clk5bdx86000b3b6qhjl9te42%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clk5bdx86000f3b6qzmzlz8sr%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clk5bdx86000d3b6qjfz7ut6m%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B50%252C50%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clk5bdx86000b3b6qhjl9te42%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clk5bdx86000a3b6q2x13bluh%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252FREADME.md%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%255D%252C%2522id%2522%253A%2522clk5bdx86000b3b6qhjl9te42%2522%252C%2522activeTabId%2522%253A%2522clk5bdx86000a3b6q2x13bluh%2522%257D%252C%2522clk5bdx86000d3b6qjfz7ut6m%2522%253A%257B%2522id%2522%253A%2522clk5bdx86000d3b6qjfz7ut6m%2522%252C%2522activeTabId%2522%253A%2522clk5be2ij01093b6qm9o4yq5i%2522%252C%2522tabs%2522%253A%255B%257B%2522type%2522%253A%2522TASK_PORT%2522%252C%2522taskId%2522%253A%2522dev%2522%252C%2522port%2522%253A3000%252C%2522id%2522%253A%2522clk5be2ij01093b6qm9o4yq5i%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522path%2522%253A%2522%252F%2522%257D%255D%257D%252C%2522clk5bdx86000f3b6qzmzlz8sr%2522%253A%257B%2522id%2522%253A%2522clk5bdx86000f3b6qzmzlz8sr%2522%252C%2522activeTabId%2522%253A%2522clk5be03d00pe3b6qglfuvdjg%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clk5bdx86000e3b6qip7azb4i%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TERMINAL%2522%252C%2522shellId%2522%253A%2522clk5bdxbi0002fnf1bk7v1h1q%2522%257D%252C%257B%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522dev%2522%252C%2522id%2522%253A%2522clk5be03d00pe3b6qglfuvdjg%2522%252C%2522mode%2522%253A%2522permanent%2522%257D%255D%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Atrue%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D) link should work
